snuba/admin/views.py

```diff
@@ -441,8 +442,58 @@ def clickhouse_trace_query() -> Response:
         )
 
     try:
-        result = run_query_and_get_trace(storage, raw_sql)
-        return make_response(jsonify(asdict(result)), 200)
+        if not raw_sql.endswith("SETTINGS log_profile_events=1"):
+            raw_sql += " SETTINGS log_profile_events=1"
+
+        query_trace = run_query_and_get_trace(storage, raw_sql)
+
+        profile_events_raw_sql = "SELECT ProfileEvents FROM system.query_log WHERE query_id = '{}' AND type = 'QueryFinish'"
+        hosts_ports_query_ids_nodes = parse_trace_for_query_ids(query_trace, storage)
+
+        for host_port_query_id_node in hosts_ports_query_ids_nodes:
+            sql = profile_events_raw_sql.format(host_port_query_id_node["query_id"])
+            system_query_result, counter = None, 0
+
+            while counter < 10:
+                # There is a race between the original query to trace finishing and the system query log populating.
+                # Sleep between the query executions.
+                system_query_result = run_system_query_on_host_with_sql(
+                    host_port_query_id_node["host"],
+                    int(host_port_query_id_node["port"]),
+                    storage,
+                    sql,
+                )
+                if not system_query_result.results:
+                    time.sleep(1)
+                    counter += 1
+                else:
+                    break
+
+            assert (
```

Comment:
Can you handle this error a little better? I think right now this will end up returning a generic 500 error to the frontend with the message `Must get ProfileEvents!`. Maybe return a timeout error or something like that?